auth/handler.go

```diff
@@ -221,7 +223,20 @@ func (h *AuthenticationHandler) Authenticate(username string, password string) (
 		}
 	}
 
-	body, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode == 202 {
+		h.logger.Infof("user %s has given correct credentials, but additional authentication factor is required", username)
+		body, _ := io.ReadAll(resp.Body)
+		var unmarshalledBody map[string]interface{}
+
+		if err := json.Unmarshal(body, &unmarshalledBody); err != nil {
+			return nil, err
+		}
```

Comment:
This should work without the intermediate `io.ReadAll` (and ignoring that one's error return value 😉):

```suggestion
		var unmarshalledBody map[string]interface{}

		if err := json.NewDecoder(resp.Body).Decode(unmarshalledBody); err != nil {
			return nil, err
		}
```

Reply:
oops, actually copied that section from somewhere else, but very true, forgot about the possibility of using Decoder API